### ISSUE TITLE
The released training code is right?

### ISSUE THREAD

**Issue by ktxu1224** [yohanshin/WHAM#98]
Hey, @yohanshin 
Thank you for releasing the training code,  however I found some problems about the training code , and I can't get the right inference result using my own training model. The problem is as follow:
1)  About the released training code, as follow, before the trajectory_refiner, whether need to update the feet_world field in output, since the root position has been changed in reset_root_velocity function, but the released code is not,  right?
![wham-refiner](https://github.com/yohanshin/WHAM/assets/22880160/937e1f07-45ef-4722-b727-275e075bd8ec)

2) I tried to retrain the model using the code in main branch, with the same config and amass.pth in stage1, but I can get the wrong result, the result is as follow, so I wonder whether there some bug in training code , or whether i should using the code in train branch to retrain model. 
![image](https://github.com/yohanshin/WHAM/assets/22880160/2def41f2-5e7e-45c3-b00c-f59704001cca)

Looking forwards to your reply, thx a lot. @yohanshin

**Comment by ktxu1224** [comment]
@dalgu90 @Arthur151 @yohanshin @RohaanA

**Comment by yohanshin** [comment]
Hi @ktxu1224 ,

1. Thank you for pointing this out. I think your point is also valid, but I intentionally scripted this refinement process as it is. The refinement process first does "coarse" refinement (`reset_root_velocity`) and then finely updates through the learning mechanism. The network's objective is to update coarsely refined trajectory (which only minimizes the foot sliding) and return feasible and smooth human trajectory. To this end, I think the network needs the information on the initial trajectory (initial feet). However, I am confident that your method will also achieve comparable performance.

2. Did you observe if your training loss / validation scores are going down? Your results seem to be incredibly bad, and I suspect that 1) your training code bursts somewhere, or 2) you did not load your checkpoint for the evaluation.

**Comment by ktxu1224** [comment]
hi, @yohanshin 

About the training result, I want to mention that, 
1) I used the data (amass.pth) and loss weight you provided without any modify, besides (1) using the torch.nn.DataParallel function to train model parallerly , (2) process data without camera motion . And also I  evaluate on the 3dpw dataset, the loss and evaulation error is as follow for you reference.
![image](https://github.com/yohanshin/WHAM/assets/22880160/ca87846b-3fd7-406e-bb2b-9b7300c73935)
loss visualization results:
![image](https://github.com/yohanshin/WHAM/assets/22880160/9c35d457-8171-4e5d-8a92-ebcba17b4224)
![image](https://github.com/yohanshin/WHAM/assets/22880160/cf8c77ac-be00-4a36-8161-b41573939b9f)
![image](https://github.com/yohanshin/WHAM/assets/22880160/dedbcf90-1edb-4071-b994-144de9ae2f7c)
![image](https://github.com/yohanshin/WHAM/assets/22880160/2fbb0168-4085-4458-af66-63699a8435b8)

I think maybe there are some bug in my code ,since validation score is going up, but the loss is seems to right, Can you give me some advice, thanks a lot.

**Comment by ktxu1224** [comment]
@yohanshin @dalgu90 @Arthur151 @RohaanA
